iox_query_influxql/src/plan/planner.rs

```diff
@@ -1053,6 +1060,129 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
             None
         })
     }
+
+    /// Expand tables from `FROM` clause in metadata queries.
+    fn expand_tables(&self, from: Option<ShowFromClause>) -> Result<Vec<String>> {
+        match from {
+            None => {
+                let mut tables = self
+                    .s
+                    .table_names()
+                    .into_iter()
+                    .map(|s| s.to_owned())
+                    .collect::<Vec<_>>();
+                tables.sort();
+                Ok(tables)
+            }
+            Some(from) => {
+                let all_tables = self.s.table_names().into_iter().collect::<HashSet<_>>();
+                let mut out = HashSet::new();
+                for qualified_name in from.iter() {
+                    if qualified_name.database.is_some() {
+                        return Err(DataFusionError::NotImplemented(
+                            "database name in from clause".into(),
+                        ));
+                    }
+                    if qualified_name.retention_policy.is_some() {
+                        return Err(DataFusionError::NotImplemented(
+                            "retention policy in from clause".into(),
+                        ));
+                    }
+                    match &qualified_name.name {
+                        MeasurementName::Name(name) => {
+                            let name = name.as_str();
+                            if all_tables.contains(name) {
+                                out.insert(name);
+                            }
+                        }
+                        MeasurementName::Regex(regex) => {
+                            for name in &all_tables {
+                                let regex = parse_regex(regex)?;
```

Comment:
```suggestion
                            let regex = parse_regex(regex)?;
                            for name in &all_tables {
```

This can be moved this outside the loop, so it isn't parsed each iteration.

Reply:
fixed